src/main/java/de/qabel/desktop/hockeyapp/VersionClient.java

```diff
@@ -0,0 +1,134 @@
+import static com.amazonaws.services.simpleworkflow.flow.junit.AsyncAssert.assertEquals;
+import static org.spongycastle.crypto.tls.ConnectionEnd.client;
+
+public class VersionClient {
+
+    private final String appId;
+    public String appVersion;
+
+    private final String baseUri;
+    private final String securityToken;
+    public HttpClient httpClient;
+
+    private List<HockeyAppVersion> versions = new LinkedList<HockeyAppVersion>();
+    private HockeyAppVersion version;
+
+    public VersionClient(HockeyAppClient hockeyAppClient) {
+        this.httpClient = hockeyAppClient.getHttpClient();
+        this.appVersion = hockeyAppClient.getAppVersion();
+        this.baseUri = hockeyAppClient.getBaseUri();
+        this.securityToken = hockeyAppClient.getSecurityToken();
+        this.appId = hockeyAppClient.getAppId();
+    }
+
+    String buildApiUri(String apiCallPath){
+        return baseUri + apiCallPath;
+    }
+
+
+    public List<HockeyAppVersion> getVersions() {
+        return versions;
+    }
+
+    public void setVersions(List<HockeyAppVersion> versions) {
+        this.versions = versions;
+    }
+
+    void loadVersions() throws IOException {
+        List<HockeyAppVersion> versions = new LinkedList<>();
+
+        HttpGet httpGet = new HttpGet(buildApiUri("/app_versions"));
+        httpGet.addHeader("X-HockeyAppToken", securityToken);
+        HttpResponse response = httpClient.execute(httpGet);
+        String responseContent = EntityUtils.toString(response.getEntity());
+
+        JSONObject parsedJson = new JSONObject(responseContent);
+        JSONArray versionArray = parsedJson.getJSONArray("app_versions");
+
+        for (int i=0; i < versionArray.length(); i++){
+            JSONObject jsonObj = versionArray.getJSONObject(i);
+            int versionId = jsonObj.getInt("version");
+            String shortVersion = jsonObj.getString("shortversion");
+            versions.add(new HockeyAppVersion(versionId, shortVersion));
+        }
+        setVersions(versions);
+    }
+
+    /**
+     * Sets version by given "shortVersion"
+     * @param shortVersion (ex.: "1.1")
+     */
+    void findAndLoadVersion(String shortVersion) throws VersionNotFoundException {
+        getVersions().forEach(version -> {
+            if(shortVersion.equals(version.getShortVersion())){
+                setVersion(version);
+            }
+        });
+        if(getVersion() == null){
+            throw new VersionNotFoundException("No Version with the shortversion: '" + shortVersion + "' found in HockeyApp");
+        }
+
+    }
+
+    public void setVersion(HockeyAppVersion version) {
+        this.version = version;
+    }
+
+    public HockeyAppVersion getVersion() {
+        return version;
+    }
+
+    public void createVersion(String version) throws IOException {
+        HttpPost request = new HttpPost(buildApiUri("/app_versions/new"));
+        request.addHeader("X-HockeyAppToken", this.appId);
+
+        //build parameters
```

Comment:
If you feel inclined to write an explaining comment, sometimes it is better to just extract that part to a method with a descriptive name.